spec/system/s3_uploads_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+describe "Uploading files to S3", type: :system do
+  fab!(:current_user) { Fabricate(:admin) }
+
+  before do
+    # setup_s3
+    # stub_s3_store
+    SiteSetting.enable_s3_uploads = true
+
+    # SiteSetting.s3_region = "us-west-1"
+    SiteSetting.s3_upload_bucket = "discoursetestbucket"
+    SiteSetting.enable_upload_debug_mode = true
+
+    SiteSetting.s3_access_key_id = "minioadmin"
+    SiteSetting.s3_secret_access_key = "minioadmin"
+    SiteSetting.s3_endpoint = "http://127.0.0.1:9000"
```

Comment:
This is key, we need to use this endpoint override instead of the bucket + region subdomain combination.